offer-maker/Direct.md

```diff
@@ -0,0 +1,17 @@
+# Direct
+<!-- FIXME: We have another md file "Direct offers" which could be mixed up with this -->
+Direct is an abstract implementation of [MangroveOffer](mangrover-offer.md), if you don't have a good understanding of MangroveOffer we recommend reading that page first.
+
+Direct should be seen as an implementation that only works for one user, the admin of the contract. This means most calls are guarded, so only the admin can call them. Having only one user, simplifies the uses of the contract, since there is no need to keep track of who is posting offer, updating offers or retracting offers, it is always the admin.
+
+Direct does many of the same things as MangroveOffer with a few key differences.
+
+**How inbound tokens are deposited to the maker:** After the funds have been transferred from the taker to the Direct contract, it chooses to leave received funds on the contract's balance, until `makerPosthook` is called. The reason for leaving the funds on the contract is that multiple offers posted by this contract may be taken in the same market order. This will result in a cumulative amount of inbound tokens being on the contract's balance. Instead of transferring them during each call to `makerExecute`, it is cheaper in gas to wait until after all offers are taken, and then transfer all the funds left on the contract to the reserve.
+
+**How outbound tokens are deposited to the taker:** When transferring the funds from the contract to the taker, Direct first tries to check if it itself has the funds, otherwise tries to get the funds from the reserve either by using a router or by using the contract itself. This means that if the contract already has the funds, no extra transfers are needed.
+
+**Approvals:** When Direct checks if the maker is allowed to set a given address as the reserve, it just checks whether the caller is the admin of the contract. Since the contract can only be used by the admin, there is no need to check anything else.
```

Comment:
One should also add a section about offer maker reserves. In a nustshell an offer of a strat is an address that has offer management rights (offer creation, retracting or updating). For `Direct` strats, there is a unique offer manager, which is also the admin of the contract. For `Forwarder` strats, any address can be an offer maker.
A `reserve` is associated to each offer maker. By default the reserve of an offer maker is the offer maker's address. The reserve is where the outbound tokens will be fetched, and where the inbound tokens will be deposited. Advanced routers may use complex protocols, such as AAVE, as reserve.
It is possible to change the default reserve of an offer maker. In `Direct` contracts this can be done by the admin without constraint, by calling the function `setReserve(admin_address, new_reserve)`. In `Forwarder` contracts, the reserve needs to call `approveMakerPooledMaker(offer_maker_address)` prior to the offer maker calling `setReserve`.

Reply:
I've added an Approvals section on the MangroveOffer page. Since approvals are basically the same for Direct and Forwarder i placed it there.

For the reserve description, i've added more to the MangroveOffer page, and made small changes to Direct and Forwarder pages.